### PR TITLE
Programmatic ownership transfer of ownables in deployer

### PR DESCRIPTION
**Pull request by asaj**
### Description

Moves the Deployer classes to detect ownable contracts programmatically when doing ownership transfer

### CODE REVIEW

File: typescript/sdk/src/gas/HyperlaneIgpDeployer.ts [hunk 1]
```diff
@@ -133,15 +133,23 @@ export class HyperlaneIgpDeployer extends HyperlaneDeployer<
...
       storageGasOracle,
       interchainGasPaymaster,
       defaultIsmInterchainGasPaymaster: overheadIgp,
     };
+    // Do not transfer ownership of StorageGasOracle
+    const ownables = await ownableContracts(contracts);
+    const filteredOwnables = pick(
+      ownables,
+      Object.keys(contracts).filter((name) => name !== 'storageGasOracle'),
+    );
```
Comment (yorhodes) [inline typescript/sdk/src/gas/HyperlaneIgpDeployer.ts:147]: why?

Comment (yorhodes) [reply typescript/sdk/src/gas/HyperlaneIgpDeployer.ts:147]: add a more descriptive note

**Review by yorhodes — APPROVED** [review]
thanks for doing this!

File: typescript/sdk/src/contracts.ts [hunk 2]
```diff
@@ -108,3 +109,21 @@ export function connectContractsMap<F extends HyperlaneFactories>(
     connectContracts(contracts, multiProvider.getSignerOrProvider(chain)),
   );
 }
+
+export async function ownableContracts(
```
Comment (jmrossy) [inline typescript/sdk/src/contracts.ts:113]: Suggest a more descriptive name like `getOwnableContracts` or `filterUnownableContracts`

**Review by jmrossy — APPROVED** [review]

**Pull request merged**